### DISCUSSION TITLE
Error creating bridge interface vibr0: operation not permitted

### DISCUSSION THREAD

**Discussion by Satlinker** [virt-manager/virt-manager#647] [Q&A]
Manjaro-Linux with win10 VM
Details:
Fehler beim Erzeugen des virtuellen Netzwerks: error creating bridge interface virbr0: Operation not permitted
Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 72, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/createnet.py", line 428, in _async_net_create
    netobj.create()
  File "/usr/lib/python3.11/site-packages/libvirt.py", line 3553, in create
    raise libvirtError('virNetworkCreate() failed')
libvirt.libvirtError: error creating bridge interface virbr0: Operation not permitted

**Comment by phrdina** [comment]
Hi, are you using `User session` or `System session`? If you right-click in virt-manager on the connection you are using and select details check what `Libvirt URI` is there. If the URI is `qemu:///session` libvirt is running as normal user and doesn't have permissions to create new network bridge which is expected.

User session has limited functionality as it doesn't have root permissions.

**Reply by Satlinker** [reply]
libviert-address: qemu:///session

If I open sudo virt-manager, I get an error:
Unable to connect to libvirt qemu:///system.
Socket-Erstellung zu '/var/run/libvirt/virtqemud-sock' fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Libvirt URI is: qemu:///system
Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/connection.py", line 923, in _do_open
    self._backend.open(cb, data)
  File "/usr/share/virt-manager/virtinst/connection.py", line 171, in open
    conn = libvirt.openAuth(self._open_uri,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libvirt.py", line 147, in openAuth
    raise libvirtError('virConnectOpenAuth() failed')
libvirt.libvirtError: Socket-Erstellung zu '/var/run/libvirt/virtqemud-sock' fehlgeschlagen: Datei oder Verzeichnis nicht gefunden

**Reply by phrdina** [reply]
You don't have to run virt-manager with sudo in order to connect to system connection. Usually you just need to add your user into libvirt group or similar.

Virt-manager is just GUI talking to libvirt. You can run virt-manager as your user but you need to use correct connection from that virt-manager. Session connection has limited functionality and by default in virt-manager it is named as `QEMU/KVM User session`. System connection is connecting to libvirt running as root and you have full functionality and in virt-manager is usually named as `QEMU/KVM`.

Seems like you don't have enabled system libvirt daemons by default. If you follow this guide https://wiki.manjaro.org/index.php/Virt-manager it should help you to start using virt-manager with system connection.

**Reply by Satlinker** [reply]
Pavel, thank you very much for your help, but I have no idea what you are talking about. I have no idea how to get this to work after hours of reading. And I am not a computer novice. I can't get the permissions, according to your link virt-manager should "prompt me for sudoer's password", which it doesn't do. I have also tried [this description](https://theduckchannel.github.io/post/how-to-install-virt-manager-on-manjaro/) with no success. I think I'll abort the VM experiment.

**Reply by phrdina** [reply]
That error message means the is no UNIX socket that virt-manager can connect to which means there is no libvirt daemon running. You need to check if libvirtd or virtqemud are enabled, you can do that by running `systemctl status libvirtd.socket` or `systemctl status virtqemud.socket`. If both are disabled and inactive that means there is no system daemon running.

As per the link you posted or the one I posted you need to enable libvirt daemon. You can do that by running `sudo systemctl enable --now libvirtd.socket`. That should enable and start the libvirtd.socket and the error message about '/var/run/libvirt/virtqemud-sock' should go away.

**Reply by Satlinker** [reply]
libvirtd.socket - libvirt legacy monolithic daemon socket
     Loaded: loaded (/usr/lib/systemd/system/libvirtd.socket; enabled; preset: disabled)
     Active: active (running) since Wed 2024-03-27 18:46:03 CET; 30min ago
virtqemud.socket - libvirt QEMU daemon socket
     Loaded: loaded (/usr/lib/systemd/system/virtqemud.socket; disabled; preset: disabled)
     Active: inactive (dead)

I cannot start libqemud.service  service does not exist
I am in the group 954(libvirt)
Libvirt address is still qemu:///session, that's probably why I cannot add networks

**Reply by gbernaldo82** [reply]
Hi, I think I have exactly the same problem. I'm trying to use the bridge mode on a qemu machine but, when I start the machine I always get the next error: 

Error al iniciar dominio: No se puede añadir puente enp1s0 puerto vnet12: La operación no está soportada

Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 72, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 108, in tmpcb
    callback(*args, **kwargs)
  File "/usr/share/virt-manager/virtManager/object/libvirtobject.py", line 57, in newfn
    ret = fn(self, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/virt-manager/virtManager/object/domain.py", line 1402, in startup
    self._backend.create()
  File "/usr/lib64/python3.12/site-packages/libvirt.py", line 1373, in create

I'm on Fedora Core 39, the guest system is RHEL8

Anyone has any idea about how to solve it? I've been with this problem a lot of time and I don't know how to solve it. I tried looking for an answer on google, stackoverflow, but nothing!

My user is part of the kvm and libvirt groups.

Thank you so much in advance!

PD: I haven't open a new thread because I think this thread is related to my problem. If that's not the case, please, let me know about it and I will open a new thread.

**Reply by phrdina** [reply]
Hi @gbernaldo82, your issue is most likely different. When configuring network device for VM if you pick `Bridge device...` as `Network source` it expect that you specify network bridge that was already created by you. Based on the name `en1s0` you are using existing network device on your host. Unless you have specific needs to use bridge I would suggest using the default NAT based network.

**Reply by phrdina** [reply]
Hi @Satlinker, correct, if your URI is `qemu:///session` it runs everything as your user and that by default cannot create the `virbr0` interface. If you want to run VM using system connection and you don't have it configured in your virt-manager you need to add new connection (click on File -> Add Connection...) and pick `QEMU/KVM` from the list of hypervisors which uses `qemu:///system`.
